### ISSUE TITLE
Error While building Dockerfile, sprockets requires Ruby version >= 2.5.0.

### ISSUE THREAD

**Issue by saikatharryc** [judge0/api#113]
While building Dockerfile it fails saying :
```
 ERROR:  Error installing rails:
   sprockets requires Ruby version >= 2.5.0.
```
coming from  [Dockerfile#L18](https://github.com/judge0/api/blob/master/Dockerfile#L18).
and I guess we can change it in [Dockerfile#L14](https://github.com/judge0/api/blob/master/Dockerfile#L14) and add new version which is equal or above 2.5.0 in [here](https://github.com/judge0/api-base/blob/master/Dockerfile#L77)

**Comment by lethehaiau** [comment]
When following the dev setup, running ./script/run-server return error:
``bin_path': can't find gem bundler (>= 0.a) (Gem::GemNotFoundException)`
I try to reinstall rails and get sem error as @saikatharryc

**Comment by lethehaiau** [comment]
It works now with judge0/api:1.4.0 not lastest : d

**Comment by saikatharryc** [comment]
@lethehaiau  are you able to run the production one?

**Comment by lethehaiau** [comment]
@saikatharryc  the development, I docker pull judge0/api:1.4.0 and update Dockerfile.dev from judge0/api to judge0/api:1.4.0 too. It works now

**Comment by hermanzdosilovic** [comment]
You can build production build with `./scripts/prod-build` that includes `--cache-from` flag that solves this problem.

Let me know if this helped you.

**Comment by noisytoken** [comment]
I tried executing  `./scripts/prod-build` in production today and got this error.

```bash
Successfully installed rack-test-0.6.3
Successfully installed actionpack-5.0.0
Successfully installed bundler-1.15.4
Building native extensions.  This could take a while...
Successfully installed pg-0.18.0
2 gems installed
The command '/bin/sh -c echo "gem: --no-document" > /root/.gemrc &&     gem install       rails:5.0.0       bundler:1.15.4       pg:0.18 &&     npm install -g aglio@2.3.0' returned a non-zero code: 1
```

**Comment by hermanzdosilovic** [comment]
On what branch did you run this?

**Comment by noisytoken** [comment]
I am on `master`. But I also tried it on [1.3.1](https://github.com/judge0/api/tree/v1.3.1)

**Comment by hermanzdosilovic** [comment]
Please do the following:
```bash
docker pull judge0/api:1.4.0
docker build -t judge0/api --cache-from judge0/api:1.4.0 . # Or ./scripts/prod-build from master branch.
```

**Comment by noisytoken** [comment]
when I run the second command on Ubuntu 18.04. I get this error:

```bash
Fetching ffi 1.11.1
Installing ffi 1.11.1 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /opt/.gem/gems/ffi-1.11.1/ext/ffi_c
/usr/local/ruby-2.3.3/bin/ruby -r ./siteconf20191106-6-1ckb066.rb extconf.rb
Cannot allocate memory - /usr/local/ruby-2.3.3/bin/ruby -r
./siteconf20191106-6-1ckb066.rb extconf.rb 2>&1

Gem files will remain installed in /opt/.gem/gems/ffi-1.11.1 for inspection.
Results logged to
/opt/.gem/extensions/x86_64-linux/2.3.0-static/ffi-1.11.1/gem_make.out

An error occurred while installing ffi (1.11.1), and Bundler cannot continue.
Make sure that `gem install ffi -v '1.11.1' --source 'https://rubygems.org/'`
succeeds before bundling.

In Gemfile:
  listen was resolved to 3.1.5, which depends on
    rb-inotify was resolved to 0.10.0, which depends on
      ffi
The command '/bin/sh -c RAILS_ENV=production bundle' returned a non-zero code: 5
```

**Comment by noisytoken** [comment]
> Please do the following:
> 
> ```shell
> docker pull judge0/api:1.4.0
> docker build -t judge0/api --cache-from judge0/api:1.4.0 . # Or ./scripts/prod-build from master branch.
> ```

Any update?

**Comment by hermanzdosilovic** [comment]
I cannot reproduce this. On clean master branch with the above commands I can build Judge0 API. But it seems to me that you have another problem that is not related to Judge0 API.
> Cannot allocate memory - /usr/local/ruby-2.3.3/bin/ruby -r

I am not really sure how to "fix" this or how to help you with this. Can you try building it on some other machine?

**Comment by saikatharryc** [comment]
@hermanzdosilovic I'm able to build by making [this](https://github.com/judge0/api/blob/master/Dockerfile#L18) from  `5.0.0` to `3.2.12`, so far it's running fine though.

**Comment by hermanzdosilovic** [comment]
This problem is now resolved with [Judge0 API v1.5.0](https://github.com/judge0/api/blob/master/CHANGELOG.md#v150-2020-01-01).
